src/client/datascience/kernel-launcher/jupyterPaths.ts

```diff
@@ -38,6 +40,15 @@ export class JupyterPaths {
         );
     }
 
+    /**
+     * Contains the name of the directory where the Jupyter extension will temporary register Kernels when using non-raw.
+     * (this way we don't register kernels in global path).
+     */
+    public async getKernelSpecTempRegistrationFolder() {
+        const dir = path.join(this.context.extensionUri.fsPath, 'temp', 'jupyter', 'kernels');
```

Comment:
All kernelspecs are always in extension folder.
This way when ever we update the extension, we always end up creating these.

I kind of like this, if users delete environments or the like, the kernels will not live around unnecessarily.

Note: If you upgrade to a new version of the extension, the exact same kernel (with the same name is created)
I.e. the algorithm for geneating kernel names hasn't changed.